src/System/Requirement/VariableFilesProtectedPath.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\System\Requirement;
+
+/**
+ * @since 10.0.4
+ */
+final class VariableFilesProtectedPath extends AbstractRequirement
+{
+    /**
+     * Constants defining directories to check.
+     *
+     * @var string[]
+     */
+    private $directories_constants;
+
+    /**
+     * Constant defining variable root directory.
+     *
+     * @var string
+     */
+    private $var_root_constant;
+
+    /**
+     * Web root directory.
+     *
+     * @var string
+     */
+    private $web_root_directory;
+
+    /**
+     * @param array $directories_constants  Constants defining directories to check.
+     * @param string $var_root_constant     Constant defining variable root directory.
+     * @param string $root_directory        Web root directory.
+     */
+    public function __construct(
+        array $directories_constants,
+        string $var_root_constant = 'GLPI_VAR_DIR',
+        string $web_root_directory = GLPI_ROOT
+    ) {
+        $this->title = __('Safe path for variables files directories');
```

Comment:
Globally, it may make more sense to call these directories "data directories". Otherwise, I think "variables" should at least be "variable" since it is used as an adjective and not a noun.

Reply:
I will change to `data directories`.